### PR TITLE
Updated setup.cfg with check-manifest

### PR DESCRIPTION
**Pull request by Sanketh7**
Adding `check-manifest` as hook for `pre-commit`

### CODE REVIEW

**Comment by taylorfturner** [comment]
@Sanketh7 needs an `update branch` on this FYI

File: setup.cfg [hunk 1]
```diff
@@ -10,3 +10,22 @@ include_trailing_comma=True
 force_grid_wrap=0
 use_parentheses=True
 line_length=88
+
+[check-manifest]
```
Comment (taylorfturner) [inline setup.cfg:14]: will also need to add `check-manifest` hook in `[.pre-commit-config.yaml](https://github.com/capitalone/DataProfiler/blob/main/.pre-commit-config.yaml)`

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+recursive-include dataprofiler *.html
+recursive-include dataprofiler *.json
+recursive-include dataprofiler *.md
+recursive-include dataprofiler *.parq
+recursive-include dataprofiler *.parquet
+recursive-include dataprofiler *.py
+recursive-include dataprofiler *.txt
+recursive-include examples *.ipynb
+recursive-include examples *.md
+recursive-include examples *.png
```
Comment (taylorfturner) [inline MANIFEST.in:17]: concerned these are included in the installation... they shouldn't be

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -45,3 +45,13 @@ repos:
...
+  # Check-manifest: ensures required non-Python files are included in MANIFEST.in
+  # https://github.com/mgedmin/check-manifest/blob/master/.pre-commit-hooks.yaml
+  - repo: https://github.com/mgedmin/check-manifest
+    rev: "0.48"
+    hooks:
+      - id: check-manifest
+        additional_dependencies: ['h5py', 'wheel', 'future', 'numpy', 'pandas',
+        'python-dateutil', 'pytz', 'six', 'pyarrow', 'chardet', 'fastavro',
+        'python-snappy', 'charset-normalizer', 'psutil', 'scipy', 'requests',
+        'networkx','typing-extensions']
```
Comment (taylorfturner) [inline .pre-commit-config.yaml:57]: run `check-manifest` on `pre-commit run`

File: MANIFEST.in (hunk 2 above)
Comment (taylorfturner) [reply MANIFEST.in:17]: that is `ipynb`, `md`, and `png`

File: setup.cfg [hunk 4]
```diff
@@ -10,3 +10,30 @@ include_trailing_comma=True
...
+ignore-default-rules=True
+ignore =
+    DataProfiler.egg-info/**
+    dataprofiler/reports/.mypy_cache/**
+    dataprofiler/tests/data/json/honeypot
+    dataprofiler/labelers/embeddings/glove-reduced-64D.txt
+    resources/__pycache__/__init__.cpython**
```
Comment (taylorfturner) [inline setup.cfg:21]: had to add these for `check-manifest` to pass

File: MANIFEST.in (hunk 2 above)
Comment (taylorfturner) [reply MANIFEST.in:17]: if I take them out I receive 
```
missing from sdist:
  examples/DL-Flowchart.png
  examples/add_new_model_to_data_labeler.ipynb
  examples/column_name_labeler.ipynb
  examples/data_profiler_demo.ipynb
  examples/data_readers.ipynb
  examples/graph_data_demo.ipynb
  examples/great_expecations/README.md
  examples/great_expecations/ge_expect_column_value_confidence_for_data_label.ipynb
  examples/great_expecations/ge_expect_column_values_vs_profile_min.ipynb
  examples/great_expecations/ge_expect_profile_numeric_columns_diff_in_range.ipynb
  examples/great_expecations/ge_expect_profile_numeric_columns_percent_diff_in_range.ipynb
  examples/intro_data_profiler.ipynb
  examples/labeler.ipynb
  examples/merge_profile_list.ipynb
  examples/popmon_dp_loader_example.ipynb
  examples/regex_labeler_from_scratch/DataLabeler_from_scratch.ipynb
  examples/structured_profilers.ipynb
  examples/unstructured_profilers.ipynb
```
and `check-manifest` recommends adding those 3 lines (15 - 17)

File: requirements-dev.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
+check-manifest>=0.48
```
Comment (taylorfturner) [inline requirements-dev.txt:1]: adding new dependency of `check-manifest` for `pre-commit`

File: setup.cfg (hunk 4 above)
Comment (taylorfturner) [reply setup.cfg:21]: that is lines 17 - 21

File: .pre-commit-config.yaml (hunk 3 above)
Comment (taylorfturner) [inline .pre-commit-config.yaml:57]: have to add these for `check-manifest` to run in the `pre-commit` environment

Comment (taylorfturner) [reply .pre-commit-config.yaml:57]: read [this](https://github.com/pre-commit/pre-commit/issues/730) but was reading a post from the creator of `pre-commit` that using `-r requirements.txt` in the `add_dependencies` is not good practice

**Review by JGSweets — APPROVED** [review]

**Review by ksneab7 — APPROVED** [review]

**Pull request merged**